descartes_core/include/descartes_core/robot_model.h

```diff
@@ -129,7 +129,12 @@ class RobotModel
    * @return                 [description]
    */
   virtual bool isValidMove(const std::vector<double> &from_joint_pose, const std::vector<double> &to_joint_pose,
-                           double dt) const = 0;
+                           double dt) const
+  {
+    return isValidMove(from_joint_pose.data(), to_joint_pose.data(), dt);
+  }
+
+  virtual bool isValidMove(const double* s, const double* f, double dt) const = 0;
```

Comment:
This method seems dangerous.  What guarantees that the data in `s` and `f` are valid (you should check for NULL).  You could literally pass any data to the method.  You are assuming that the data is the size of the DOF of the robot model.  The reason for this method is because you have chosen to represent joint solutions as raw memory(am I wrong)?  Does this really save you that much time/memory?  If you kept the joint solution representation as a vector, and the group of joint solutions as a vector of vectors, isn't that a reasonably compact representation.